content/spin/sqlite-api-guide.md

```diff
@@ -0,0 +1,150 @@
+
+| Operation  | Parameters | Returns | Behavior |
+|------------|------------|---------|----------|
+| `open`  | name | connection  | Open the database with the specified name. If `name` is the string "default", the default database is opened, provided that the component that was granted access in the component manifest from `spin.toml`. Otherwise, `name` must refer to a store defined and configured in a [runtime configuration file](./dynamic-configuration.md#key-value-store-runtime-configuration) supplied with the application.|
+| `query` | connection, sql, parameters | database records | Executes the SQL query and returns the results of the query. Use this for SQL statements that return results, such as SELECT. |
+| `execute` | connection, sql, parameters | - | Executes the SQL statement. Use this for SQL statements that does not return a result, such as an INSERT, UPDATE or DELETE. |
+| `close` | connection | - | Close the specified `connection`. |
+
+The exact detail of calling these operations from your application depends on your language:
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+Because the SQLite API is experimental, Rust projects must opt into it by enabling the `experimental-sqlite` SDK feature in Cargo.toml.
+
+```toml
+spin-sdk = { git = "https://github.com/fermyon/spin", tag = "v1.2.0", features = ["experimental-sqlite"] }
+```
+
+When the `experimental-sqlite` feature is enabled, SQLite functions are available in the `spin_sdk::sqlite` module. The function names match the operations above. For example:
+
+```rust
+use anyhow::Result;
+use serde::Serialize;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    sqlite::{DataTypeParam, Error, Connection},
+};
+
+#[http_component]
+fn handle_request(req: Request) -> Result<Response> {
+    // !!!!!!!!!!!!!! TODO:
+    // let connection = Connection::open_default()?;
+    let connection = Connection::open("default")?;
+
+    let execute_params = [
+        DataTypeParam::Text("Try out Spin SQLite"),
+        DataTypeParam::Text("Friday"),
+    ];
+    connection.execute(
+        "INSERT INTO todos (description, due) VALUES (?, ?)",
+        execute_params.as_slice(),
+    )?;
+
+    let rowset = connection.query(
+        "SELECT id, description, due FROM todos",
+        &[]
+    )?;
+
+    let todos = rowset.rows().map(|row|
+        ToDo {
+            id: row.get::<u32>("id").unwrap(),
+            description: row.get::<&str>("description").unwrap().to_owned(),
+            due: row.get::<&str>("due").unwrap().to_owned(),
+        }
+    );
+
+    let body = serde_json::to_vec(&todos)?;
+    Ok(http::Response::builder().status(200).body(Some(body.into()))?)
+}
+
+// Helper for returning the query results as JSON
+#[derive(Serialize)]
+struct ToDo {
+    id: u32,
+    description: String,
+    due: String,
+}
+```
+
+**General Notes** 
+* Parameters are instances of the `DataTypeParam` enum; you must wrap raw values in this type.
```

Comment:
TODO: don't want us to forget that `DataTypeParam` has been renamed `ValueParam` and `DataTypeResult` has been renamed `ValueResult`.